pytorch_lightning/trainer/supporters.py

```diff
@@ -177,3 +178,94 @@ def to_disk(self) -> None:
             # Write predictions for current file to disk
             with fs.open(filepath, "wb") as fp:
                 torch.save(outputs, fp)
+
+
+class AverageValueMeter:
+    def __init__(self):
+        super(AverageValueMeter, self).__init__()
+        self.reset()
+        self.val = 0
+
+    def add(self, value, n=1):
+        self.val = value
+        self.sum += value * n
+        if n <= 0:
+            raise ValueError("Cannot use a non-positive weight for the running stat.")
+        elif self.n == 0:
+            self.mean = 0.0 + value  # This is to force a copy in torch/numpy
+            self.std = np.inf
+            self.mean_old = self.mean
+            self.m_s = 0.0
+        else:
+            self.mean = self.mean_old + n * (value - self.mean_old) / float(self.n + n)
+            self.m_s += n * (value - self.mean_old) * (value - self.mean)
+            self.mean_old = self.mean
+            self.std = np.sqrt(self.m_s / (self.n + n - 1.0))
+        self.var = self.std ** 2
+
+        self.n += n
+
+    def value(self):
+        return self.mean, self.std
+
+    def reset(self):
+        self.n = 0
+        self.sum = 0.0
+        self.var = 0.0
+        self.val = 0.0
+        self.mean = np.nan
+        self.mean_old = 0.0
+        self.m_s = 0.0
+        self.std = np.nan
+
+
+class GradNormTracker:
+
+    def __init__(self, aggregation_mode, norm_type):
+        self._grad_norm_dic = {}
+        self.norm_type = norm_type
+
+        if aggregation_mode == 'optimizer':
+            self.name_mapping = self.aggregate_over_optimizer
+        elif aggregation_mode == 'optimizer+parameters':
+            self.name_mapping = self.aggregate_over_optimizer_and_parameters
+        elif aggregation_mode =='parameters':
+            self.name_mapping = self.aggregate_over_parameters
```

Comment:
Can you raise an exception and provide the correct option list if aggregation_mode is not one of them.